helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1578,4 +1586,193 @@ public void close() {
     }
   }
 
+  @Override
+  public boolean addResourceWithWeight(String clusterName, IdealState idealState,
+      ResourceConfig resourceConfig) {
+    // Null checks
+    if (clusterName == null || clusterName.isEmpty()) {
+      throw new HelixException("Cluster name is null or empty!");
+    }
+    if (idealState == null || !idealState.isValid()) {
+      throw new HelixException("IdealState is null or invalid!");
+    }
+    if (resourceConfig == null || !resourceConfig.isValid()) {
```

Comment:
`ResourceConfig.isValid()` always returns true. Is this expected or a bug? How to verify resourceConfig is empty?

Reply:
This is okay. I think if we need to add more logic here, we can in the future. However, that work isn't strictly in the scope of this diff.